client/src/stores/workflowConnectionStore.ts

```diff
@@ -42,82 +42,88 @@ interface TerminalToOutputTerminals {
     [index: string]: OutputTerminal[];
 }
 
-export const useConnectionStore = defineStore("workflowConnectionStore", {
-    state: (): State => ({
-        connections: [] as Array<Readonly<Connection>>,
-        invalidConnections: {} as InvalidConnections,
-        inputTerminalToOutputTerminals: {} as TerminalToOutputTerminals,
-        terminalToConnection: {} as { [index: string]: Connection[] },
-        stepToConnections: {} as { [index: number]: Connection[] },
-    }),
-    getters: {
-        getOutputTerminalsForInputTerminal(state: State) {
-            return (terminalId: string): OutputTerminal[] => {
-                return state.inputTerminalToOutputTerminals[terminalId] || [];
-            };
+export const useConnectionStore = (workflowId: string) => {
+    if (!workflowId) {
+        throw new Error("WorkflowId is undefined");
+    }
+
+    return defineStore(`workflowConnectionStore${workflowId}`, {
```

Comment:
Nice solution! Is this documented somewhere as a pattern for scoping stores ?

Reply:
I found the concept here: https://github.com/vuejs/pinia/discussions/840#discussioncomment-2784643
Seems to work very well.